src/Stratis.Features.FederatedPeg/FederatedPegOptions.cs

```diff
@@ -0,0 +1,44 @@
+﻿using Microsoft.Extensions.DependencyInjection;
+using NBitcoin;
+using Stratis.Bitcoin.Configuration;
+
+namespace Stratis.Features.FederatedPeg
+{
+    public class FederatedPegOptions
+    {
+        public IServiceCollection Services { get; }
+        public NodeSettings NodeSettings { get; }
+
+        public FederatedPegOptions(IServiceCollection services, NodeSettings nodeSettings)
+        {
+            this.Services = services;
+            this.NodeSettings = nodeSettings;
+        }
+
+        public void SetCounterChainNetwork(Network network)
+        {
+            this.Services.AddSingleton(new CounterChainNetworkHolder(network));
+        }
+
+        public void SetCounterChainNetworkSelector(NetworksSelector networksSelector)
```

Comment:
Do you think we can avoid doing this eg. by passing `this.GetCounterChainNetwork(networksSelector)` directly into `SetCounterChainNetwork`? Would like to avoid passing around `NetworksSelector`s as well as `Network`.

Reply:
+ it shouldn't not the responsibility of the options class to determine which network gets selected.

Reply:
I agree with @rowandh here, whats the issue with just passing in the Network?

Reply:
We usually don't build a node by passing in the network - we pass in a networks selector and the parameters decide which network to use.

It's the same with sidechains... we don't know which network is going to be the counter chain.

Reply:
And wait why shouldn't it be the responsibility of the options class? It's the responsibility of the NodeSettings class